app/src/ui/app.tsx

```diff
@@ -2548,6 +2579,46 @@ export class App extends React.Component<IAppProps, IAppState> {
     return selectedState.state.pullRequestState
   }
 
+  private openBypassPushProtection = (secret: ISecretScanResult) => {
+    return new Promise<IAPICreatePushProtectionBypassResponse | null>(
+      resolve => {
+        this.props.dispatcher.showPopup({
+          type: PopupType.BypassPushProtection,
+          secret,
+          bypassPushProtection: (
+            secret: ISecretScanResult,
+            reason: BypassReasonType
+          ) => {
+            this.bypassPushProtection(secret, reason)
+              .then(response => {
+                resolve(response)
+              })
+              .catch(() => {
+                resolve(null)
+              })
+              .finally(() => {
+                this.props.dispatcher.closePopup(PopupType.BypassPushProtection)
+              })
+          },
+          onDismissed: () => {
+            resolve(null)
+          },
+        })
```

Comment:
This makes it so that when user clicks "Bypass" in push protection dialog, it waits for the return of an async function set the state to "Bypassed"... but it is waiting for the user to interact with bypass dialog and the subsequent api call. Thus needing this promise. 

Another solution I thought of was storing the "bypassed state" somewhere on the repository state and updating and passing that down to the push protection dialog. 

Happy to change this if this feels wrong? like not react? I do not believe it is a pattern we have seen in the code base so I wasn't sure if it breaks a react convention of using state.